utility/src/main/java/org/oppia/util/networking/NetworkConnectionUtil.kt

```diff
@@ -0,0 +1,40 @@
+package org.oppia.util.networking
+
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.NetworkInfo
+import androidx.annotation.VisibleForTesting
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class NetworkConnectionUtil @Inject constructor(private val context: Context) {
+  enum class ConnectionStatus {
```

Comment:
Also ditto here--let's specify what this is & what each of the constants are.

Reply:
Added comments for each constant.

Reply:
Also, please add a KDoc for the enum class itself.

Reply:
Added KDoc.